### DISCUSSION TITLE
Debugging V on VSCode: VERROR_MESSAGE Header file <gc.h>, needed for module `builtin` was not found. Please install the corresponding development headers.

### DISCUSSION THREAD

**Discussion by bru65pag** [vlang/v#20226] [Questions and Answers]
I am running VSCode on Ubuntu with WSL on Windows 10. I have installed the V language support on VSCode. I also installed C/C++from microsoft. I have created a json.launch file, indicating the path to gdb:
**"configurations": [
    {
      "name": "(gdb) Launch",
      "type": "cppdbg",
      "request": "launch",
      "program": "${workspaceFolder}/a",
      "args": [],
      "stopAtEntry": false,
      "cwd": "${fileDirname}",
      "environment": [],
      "externalConsole": false,
      "MIMode": "gdb",
      "miDebuggerPath": "/usr/bin/gdb",
    }
  ]**
When I start a debugging session, I have the following error message:
**VERROR_MESSAGE Header file <gc.h>, needed for module `builtin` was not found. Please install the corresponding development headers.**
Corresponding to the following lines in a c file displayed on screen (I guess it is the c generated by the compiler):
**#if __has_include(<gc.h>)
#include <gc.h>
#else
#error VERROR_MESSAGE Header file <gc.h>, needed for module `builtin` was not found. Please install the corresponding development headers.
#endif**
I searched internet, cannot find which development headers need to be installed. Can you help? Thanks.

**Comment by JalonSolov** [comment]
You have a mix of things that don't usually work together.  Running something in WSL with Ubuntu will not pay any attention to the Windows compiler.  You'll need to install a Linux compiler in WSL.

You can follow these instructions to get the Linux compiler: https://github.com/vlang/v/wiki/Installing-a-C-compiler-on-Linux-and-macOS

Once you have that installed, try vscode again, and see if the problem is fixed.

**Reply by bru65pag** [reply]
Thanks for your suggestion. I have a full linux toolchain in WSL. I managed to get the above error fixed after installing the gc garbage collector [https://www.hboehm.info/gc/](https://www.hboehm.info/gc/) on my Ubuntu (why was is not installed with v? Should it be?). Now my gdb in VSC is failing with the the following:


**=thread-group-added,id="i1"
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".
Warning: Debuggee TargetArchitecture not detected, assuming x86_64.
=cmd-param-changed,param="pagination",value="off"
/build/gdb-ZgDh0V/gdb-12.1/gdb/psymtab.c:1142: internal-error: expand_symtabs_matching: Assertion `symtab != nullptr' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) [answered Y; input not from terminal]
/build/gdb-ZgDh0V/gdb-12.1/gdb/psymtab.c:1142: internal-error: expand_symtabs_matching: Assertion `symtab != nullptr' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Create a core file of GDB? (y or n) [answered Y; input not from terminal]
ERROR: Unable to start debugging. GDB exited unexpectedly.
The program '/home/bruno/coding/v/sandbox/a' has exited with code 42 (0x0000002a).**


Gave up on this one for the time being, and decided to give gdb outside of VSC a try. I ran into the following issue:


**Program received signal SIGSEGV, Segmentation fault.
0x000000000045e292 in GC_find_limit_with_bound ()**

Many others have the same issue (see [https://github.com/vlang/v/issues/14904](https://github.com/vlang/v/issues/14904). Apparently, one should tell the debugger to ignore SIGSEGV: 

**handle SIGSEGV nostop noprint pass**

With this, I am able to use GDB standalone. I still need to figure out why I cannot use GDB from within VSC.

Last point: [this article](https://github.com/vlang/v/wiki/Troubleshooting-(debugging)-executables-created-with-V-in-GDB) is worth reading. In particular, setting breakpoint on function names require to name them the way they are in the C code. For example, if you have a module named main and a function main, the breakpoint is main__main

**Reply by JalonSolov** [reply]
Installing libgc separately won't help, unless you also compile with `-d dynamic_boehm`.  V already comes with `thirdparty/libgc`, and that header file is in the `include` subdir under that, and that is what it normally tried to use.

I don't know why this didn't work for you.

How did you install V?  From source?  That is the recommended method.

https://github.com/vlang/v#installing-v-from-source
